src/Cli.Tests/UpdateRuntimeTests.cs

```diff
@@ -0,0 +1,132 @@
+        [TestMethod]
+        public void TestAddDepthLimitForGraphQL()
+        {
+            _fileSystem!.AddFile(_testConfigFile, new MockFileData(INITIAL_CONFIG));
+            int maxDepthLimit = 8;
+
+            Assert.IsTrue(_fileSystem!.File.Exists(_testConfigFile));
+
+            // Initial State
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(INITIAL_CONFIG, out RuntimeConfig? config));
+            Assert.IsNotNull(config);
+            Assert.IsNull(config!.Runtime!.GraphQL!.DepthLimit);
+
+            // Add Depth Limit
+            UpdateRuntimeOptions options = new(
+                depthLimit: maxDepthLimit,
+                config: _testConfigFile
+            );
+            Assert.IsTrue(TryUpdateRuntimeSettings(options, _runtimeConfigLoader!, _fileSystem!));
+
+            // Assert after adding Depth Limit
+            string updatedConfig = _fileSystem!.File.ReadAllText(_testConfigFile);
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(updatedConfig, out config));
+            Assert.IsNotNull(config);
+            Assert.IsNotNull(config!.Runtime!.GraphQL!.DepthLimit);
+            Assert.AreEqual(maxDepthLimit, config!.Runtime!.GraphQL!.DepthLimit);
+        }
+
+        /// <summary>
+        /// Test to update the current depth limit for GraphQL and removal the depth limit using -1.
+        /// </summary>
+        [DataTestMethod]
+        [DataRow(20, DisplayName = "Update current depth limit for GraphQL.")]
+        [DataRow(-1, DisplayName = "Remove depth limit from GraphQL.")]
+        public void TestUpdateDepthLimitForGraphQL(int? newDepthLimit)
+        {
+            int currentDepthLimit = 8;
+
+            // Generate Config with Depth Limit
+            RuntimeConfigLoader.TryParseConfig(INITIAL_CONFIG, out RuntimeConfig? config);
+            Assert.IsNotNull(config);
+            config = config with
+            {
+                Runtime = config!.Runtime! with
+                {
+                    GraphQL = config!.Runtime!.GraphQL! with
+                    {
+                        DepthLimit = currentDepthLimit
+                    }
+                }
+            };
+
+            _fileSystem!.AddFile(_testConfigFile, new MockFileData(config.ToJson()));
+
+            // Update Depth Limit
+            UpdateRuntimeOptions options = new(
+                depthLimit: newDepthLimit,
+                config: _testConfigFile
+            );
+            Assert.IsTrue(TryUpdateRuntimeSettings(options, _runtimeConfigLoader!, _fileSystem!));
+
+            // Assert after adding Depth Limit
+            string updatedConfig = _fileSystem!.File.ReadAllText(_testConfigFile);
+            Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(updatedConfig, out config));
+            Assert.IsNotNull(config);
+
+            int? expectedDepthLimit = newDepthLimit == -1 ? null : newDepthLimit;
+            Assert.AreEqual(expectedDepthLimit, config!.Runtime!.GraphQL!.DepthLimit);
```

Comment:
if user explictly sets depth-limit to -1, that value should be present in the config and written to disk.